### DISCUSSION TITLE
Using Docker rootless: how to connect to mail server running on another user

### DISCUSSION THREAD

**Discussion by daks** [roundcube/roundcubemail-docker#274] [Q&A] [answered]
Hi,
I'm trying to setup Roundcube with the provided docker-compose files, I'm testing with the fpm-postgres-nginx one.
I'm running it with Docker rootless with a user named 'roundcube'.
I also have my imap/smtp server running on Docker rootless with a different user named 'mail'.

Roundcube runs OK, I see the login page but can't connect, it seems that roundcube can't access the imap server. I see a log for roundcube but none on the imap side.
```
roundcubemail     | errors: <xx> IMAP Error: Login failed for user@example.com against imap.example.com from 172.128.0.1. Could not connect to ssl://imap.example.com:993: Connection refused in /var/www/html/program/lib/Roundcube/rcube_imap.php on line 211 (POST /?_task=logi
n&_action=login)                             
```

Is it possible and what sould I do to make it work?
I would prefer to keep both stacks (roundcube and mail server) isolated with two different users.

**Comment by pabzm** [comment]
It should be possible, and Roundcube doesn't share any state or data backend with the mailserver, so different users shouldn't do any difference.

~~From the error message I'd look into the logs of the mailserver next.~~

(I'm not sure if our container images are ready for rootless, but if it works I'd be happy to include a new compose-file as an example!)

**Reply by pabzm** [reply]
Sorry, I wrote faster than I read.

Are you sure the mailserver container is running? No output at all from it sounds like a fundamental problem.

If it is running properly, try if you can reach the mailserver container from the roundcube container:

```
docker compose exec roundcube curl -v -u user:password imaps://mailhost/INBOX
```

**Comment by daks** [comment]
It may be possible that rootless docker have networking restrictions that rootful don't have. In which case my setup wouldn't work at all :(

I'm still looking for answers about that

**Comment by daks** [comment, marked as answer]
After searching for hours or days, I finally found a solution to enable a container running in Docker rootless to access the host itself. I will document it here in case it may help someone else:
- use a recent version of Docker. I think it's at least version 26. For me it meant not use Debian packages but instead docker-ce ones from Docker apt repository
- install Docker rootless per user as usual
- configure rootlesskit to enable host loopback access by adding a file `~/.config/systemd/user/docker.service.d/override.conf` with
```
[Service]
Environment="DOCKERD_ROOTLESS_ROOTLESSKIT_DISABLE_HOST_LOOPBACK=false"
```
- daemon-reload systemd and restart docker service
- enjoy :)

Relevant PR in moby project https://github.com/moby/moby/pull/47352
